Jeplind Morico

jeplindm opened the pull request:
testing pull request

Commit:
Jeplind Morico

Commit:
POST, GET, PUT, PATCH, DELETE

Commit:
rev. post

Commit:
rev. documentation

dhimashary commented on models/todo.js:
```diff
@@ -0,0 +1,41 @@
+          msg: "must be in date format"
+        },
+        isAfter: {
+          args: `${new Date()}`,
```
untuk validasi tanggal tidak bisa menggunakan isAfter karena argsnya jalan ketika server dijalankan, misal server nyala di 17 agustus , di 19 agustus kita masih bisa bikin todo untuk 18 agustus (karena new date nya dibuat saat 17 agustus)

dhimashary commented on routes/index.js:
```diff
@@ -0,0 +1,17 @@
+const Controller = require('../controllers/controller')
+
+
+router.post('/todos', Controller.postTodo);
```
lebih baik dibuat ke todo.js jadi semua router todo ada disitu

dhimashary commented:
need improvement:
- pembuatan model user, validasi email format, validasi email unique
- validasi title, description, status, duedate tidak boleh kosong/null

Commit:
rev. perbaikan + register,login

Commit:
authtics,authorize,error-handler

dhimashary commented on server/controllers/user-controller.js:
```diff
@@ -0,0 +1,55 @@
+        .then((user) => {
+            if (!user) { //if user is not found
+                throw {msg: "invalid email/password", status: 401}
+            } else if (!hashedPassword(payload.password, user.password)) { //compare password fail
```
disini kita perlu nge hash password atau membandingkan password ?

dhimashary replied:
diperbaiki ya :D

dhimashary commented:
config controllers dll yang ada diluar server di hapus ya

Commit:
Jeplind Morico
1. Add client-main.js
2. Add client-index.html
3. Rev. model user

Commit:
Jeplind Morico
1. Add Sign In Google

Commit:
Jeplind Morico
1. Rev. dokumentasi
2. Handling Error pada Client
3. Rev. main.js weather function

dhimashary commented:
di client label dan input type kebalik, biasanya label dulu ya baru input type

dhimashary commented:
tombol edit delete done numpuk

dhimashary commented:
ketika edit, sebaiknya data yang sebelumnya dimasukan ke form edit sehingga user tidak perlu ketik ulang dari awal semuanya

dhimashary commented:
tanggal di client tolong di format menjadi tanggal yang mudah dibaca oleh user

dhimashary commented:
tombol register login logout muncul terus di atas, harusnya ketika sudah login, tombol login register di hide

dhimashary commented:
ketika berhasil menambah todo baiknya form dibersihkan valuenya

dhimashary commented:
tombol edit kalo dipencet beberapa kali formnya numpuk kebawah

dhimashary commented:
ketiak user delete baiknya ada konfirmasi apakah yakin mau delete / tidak

dhimashary commented:
ketika login logout login lagi cuacanya jadi dobel numpuk kebawah

dhimashary commented on server/README.md:
```diff
@@ -0,0 +1,360 @@
+        "status": "not finished",
+        "due_date": "2020-10-29T00:00:00.000Z",
+        "createdAt": "2020-10-26T10:43:44.518Z",
+        "updatedAt": "2020-10-26T10:43:44.518Z"
```
ga ada user id nya?

Commit:
deploy to heroku

Commit:
fix port and package.json

Commit:
add database production in config.json

Commit:
rev app.js

Commit:
rev. app.js

Commit:
trial config.json

Commit:
trial config.json

Commit:
remove file .env from server

Commit:
trial config and .env

Commit:
trial config.json

Commit:
rev. user controller

Commit:
try fix jwt and user controller

Commit:
fix config.json production

Commit:
fix app.js

Commit:
fix main.js client

Commit:
deploy heroku + deploy firebase
